client/server/server.go

```diff
@@ -811,26 +792,201 @@ func (s *Server) Down(ctx context.Context, _ *proto.DownRequest) (*proto.DownRes
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
+	if err := s.cleanupConnection(); err != nil {
+		log.Errorf("failed to shut down properly: %v", err)
+		return nil, err
+	}
+
+	state := internal.CtxGetState(s.rootCtx)
+	state.Set(internal.StatusIdle)
+
+	return &proto.DownResponse{}, nil
+}
+
+func (s *Server) cleanupConnection() error {
 	s.oauthAuthFlow = oauthAuthFlow{}
 
 	if s.actCancel == nil {
-		return nil, fmt.Errorf("service is not up")
+		return ErrServiceNotUp
 	}
 	s.actCancel()
 
-	err := s.connectClient.Stop()
-	if err != nil {
-		log.Errorf("failed to shut down properly: %v", err)
-		return nil, err
+	if s.connectClient == nil {
+		return nil
+	}
+
+	if err := s.connectClient.Stop(); err != nil {
+		return err
 	}
+
+	s.connectClient = nil
 	s.isSessionActive.Store(false)
 
+	log.Infof("service is down")
+
+	return nil
+}
+
+func (s *Server) Logout(ctx context.Context, msg *proto.LogoutRequest) (*proto.LogoutResponse, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	if msg.ProfileName != nil && *msg.ProfileName != "" {
+		return s.handleProfileLogout(ctx, msg)
+	}
+
+	return s.handleActiveProfileLogout(ctx)
```

Comment:
should we allow logout for other users' profiles? 

we allow "down" for other users' profiles but logout is more destructive since we delete the peer from the management.